flax/nnx/variablelib.py

```diff
@@ -747,6 +747,38 @@ class Intermediate(Variable[A]):
   pass
 
 
+class Perturbations(Variable[A]):
```

Comment:
Also, we should inherit from `Intermediate` for all temp Variables.
```suggestion
class Perturbations(Intermediate[A]):
```